src/views/shared/amount-input/amount-input.view.tsx

```diff
@@ -0,0 +1,225 @@
+      if (!isAsyncTaskCompleted(fiatExchangeRatesTask)) {
+        return 0;
+      }
+
+      const fixedTokenAmount = getFixedTokenAmount(tokensAmount.toString(), account.token.decimals);
+
+      return getTokenAmountInPreferredCurrency(
+        fixedTokenAmount,
+        account.token.USD,
+        preferredCurrency,
+        fiatExchangeRatesTask.data
+      );
+    }
+
+    /**
+     * Converts an amount in fiat to tokens.
+     */
+    function convertAmountToTokens(fiatAmount: number) {
+      const tokensAmount = fiatAmount / account.token.USD;
+
+      return parseUnits(tokensAmount.toFixed(account.token.decimals), account.token.decimals);
+    }
+
+    function updateAmountState({ tokens, fiat }: Amount, inputValue: string, showInFiat: boolean) {
+      const newAmountWithFee = tokens.add(fee);
+      const isAmountNegative = tokens.lte(BigNumber.from(0));
+      const isNewAmountWithFeeMoreThanFunds = newAmountWithFee.gt(BigNumber.from(account.balance));
+      const areFundsExceededDueToFee =
+        isNewAmountWithFeeMoreThanFunds && tokens.lte(BigNumber.from(account.balance));
+      const isAmountCompressedInvalid =
+        txType === TxType.Deposit || txType === TxType.ForceExit
+          ? false
+          : isTransactionAmountCompressedValid(tokens) === false;
+      const isAmountInvalid =
+        isAmountNegative ||
+        isNewAmountWithFeeMoreThanFunds ||
+        areFundsExceededDueToFee ||
+        isAmountCompressedInvalid;
+
+      setValue(inputValue);
+      setAmount({ tokens, fiat });
+      setIsAmountNegative(isAmountNegative);
+      setIsAmountWithFeeMoreThanFunds(isNewAmountWithFeeMoreThanFunds);
+      setIsAmountCompressedInvalid(isAmountCompressedInvalid);
+      setShowInFiat(showInFiat);
+      onChange({
+        amount: { tokens, fiat },
+        showInFiat,
+        isInvalid: isAmountInvalid,
+        areFundsExceededDueToFee,
+      });
+    }
+
+    /**
+     * Handles input change events. It's going to check that the input value is a valid
+     * amount and calculate both the tokens and fiat amounts for a value. It will also
+     * trigger the validation checks.
+     */
+    function handleInputChange(event: { target: { value: string } }) {
+      const decimals =
+        account?.token?.decimals === undefined ? MAX_TOKEN_DECIMALS : account.token.decimals;
+      const regexToken = `^\\d*(?:\\.\\d{0,${decimals}})?$`;
+      const regexFiat = `^\\d*(?:\\.\\d{0,2})?$`;
+      const INPUT_REGEX = new RegExp(showInFiat ? regexFiat : regexToken);
+
+      if (INPUT_REGEX.test(event.target.value)) {
+        if (showInFiat) {
+          const newAmountInFiat = Number(event.target.value);
+          const newAmountInTokens = convertAmountToTokens(newAmountInFiat);
+          const fixedAmountInTokens = fixTransactionAmount(newAmountInTokens);
+
+          updateAmountState(
+            { tokens: fixedAmountInTokens, fiat: newAmountInFiat },
+            event.target.value,
```

Comment:
why not use newAmountInFiat that already has the Number() filtering?

Reply:
Because this param is used to set the property `value` of the local state, and it's just the string which is going to be displayed in the input, but it's only needed for that.

Reply:
And what about to avoid leading zeroes?

Reply:
Currently we are supporting leading zeroes. The problem using the `Number()` filtering is that when you write for example `0.` it will be parsed always as `0` so you won't be able to write decimal numbers. Maybe we can improve this in another task (avoiding leading zeroes) @AlbertoElias